### PR TITLE
Enable GpuShuffledSymmetricHashJoin by default

### LINKED ISSUE #10243: Enable GpuSymmetricHashJoinExec by default
GpuSymmetricHashJoinExec added in #10240 is disabled by default.  This should be enabled by default at some point.

### PR DESCRIPTION
**Pull request by jlowe**
Fixes #10243.  Turns on the new symmetric hash join exec by default and fixes a few bugs that were found.  One was a double-close bug, and the other is a incorrect binding of a join condition expression.  This binding order bug also existed in the original GpuHashJoin code, so I fixed it there as well (and verified tests pass even if we disable symmetric hash join).

### CODE REVIEW

**Comment by jlowe** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala [hunk 1]
```diff
@@ -522,9 +522,17 @@ class ConditionalHashJoinIterator(
       withResource(GpuColumnVector.from(leftData.getBatch)) { leftTable =>
         withResource(GpuColumnVector.from(rightData.getBatch)) { rightTable =>
           val maps = joinType match {
-            case _: InnerLike =>
+            case _: InnerLike if buildSide == GpuBuildRight =>
               Table.mixedInnerJoinGatherMaps(leftKeys, rightKeys, leftTable, rightTable,
                 compiledCondition, nullEquality)
+            case _: InnerLike if buildSide == GpuBuildLeft =>
```
Comment (revans2) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala:528]: nit: Should we update the error message in the exception to include the build side?

**Review by revans2 — APPROVED** [review]
Looks good. A few nits that I don't really care about.

**Comment by jlowe** [comment]
build

**Comment by jlowe** [comment]
build

**Pull request merged**